onboarding/source/amb_light_service.c

```diff
@@ -18,22 +26,76 @@ static void lightServiceTask(void * pvParameters);
 uint8_t initLightService(void) {
     /* USER CODE BEGIN */
     // Create the task and queue here.
+    BaseType_t xReturned = pdFAIL;
+
+    if (xLightServiceHandle == NULL) {
+        xReturned = xTaskCreate(lightServiceTask,
+                                LIGHT_SERVICE_NAME,
+                                LIGHT_SERVICE_STACK_SIZE,
+                                NULL,
+                                LIGHT_SERVICE_PRIORITY,
+                                &xLightServiceHandle
+                            );
+    }
+
+    // Create queue
+    if (xQueue == NULL) {
+        xQueue = xQueueCreate(LIGHT_QUEUE_LENGTH, 
+                            LIGHT_SERVICE_ITEM_SIZE);
+    }
+
+    if(xReturned == pdFAIL) {
+        printf("Error - could not create light service task");
+    }
 
     /* USER CODE END */
+
+    
     return 1;
 }
 
 static void lightServiceTask(void * pvParameters) {
     /* USER CODE BEGIN */
     // Wait for MEASURE_LIGHT event in the queue and then print the ambient light value to the serial port.
+    adcData_t adc_data;
+    adcData_t *adc_data_ptr = &adc_data;
+
+    // Wait for MEASURE_LIGHT event in the queue and then print the ambient light value to the serial port.
+    while (1) {
+        if (xQueueReceive(xQueue, NULL, (TickType_t) 11) == pdPASS) {
+            adcStartConversion(adcREG1, 
+                                adcGROUP1);
+
+            while (!adcIsConversionComplete(adcREG1, 
+                                            adcGROUP1));
+
+            adcGetData(adcREG1, 
+                        adcGROUP1, 
+                        adc_data_ptr);
 
+            unsigned char text[2];
+
+            // LSB by masking with 0xFF
+            uint8_t first_half = (uint16_t) adc_data_ptr->value & 0xFF;
```

Comment:
`adc_data_ptr->value` returns the ADC value we're interested in. It's a `uint16_t`. It looks like you're separating the upper and lower bytes of the value, casting them to chars, and printing them. For example, imagine `adc_data_ptr->value` is 10. Then, `first_half = 0b1010` and `second_half = 0`. Then, `text[0] = '\n'` and `text[1] = '\0'` as those numbers correspond to these characters in the ASCII table. You'll be printing `\n\0` instead of `10`.

Reply:
Look into using `snprintf` instead.